plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImageWizard.java

```diff
@@ -48,6 +52,30 @@ public DeployImageWizard(IDeployImageParameters wizardModel) {
 
 	@Override
 	public void addPages() {
+	    if (getModel().originatedFromDockerExplorer()) {
+	        addPage(new ConnectionWizardPage(this, new IConnectionAware<IConnection>() {
```

Comment:
I had trouble casting a IConnectionAware<IConnection> to a IConnectionAware<Connection> so if your solution works, let's adopt it